### PR TITLE
Cristin import on sqs queue

### PR DESCRIPTION
**Pull request by anetteOlli**

### CODE REVIEW

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java [hunk 1]
```diff
@@ -313,21 +315,37 @@ void handlerThrowContributorWithoutAffiliationExceptionWhenTheCristinObjectHasCo
...
                                                                 .objectWithAffiliationWithoutRole();
         var eventBody = createEventBody(cristinObjectWithAffiliationWithoutRoles);
         var eventReference = createEventReference(eventBody);
 
-        Executable action = () -> handler.handleRequest(eventReference, outputStream, CONTEXT);
-        assertThrows(AffiliationWithoutRoleException.class, action);
+        handler.handleRequest(eventReference, CONTEXT);
+
+        var expectedExceptionName = AffiliationWithoutRoleException.class.getSimpleName();
+        UnixPath expectedFilePath = constructExpectedFilePathForEntryWithUnkownFields(eventBody,
```
Comment (torarnet) [inline cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java:342]: do you prefer typed declaration or var?

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinPatchEventConsumerTest.java [hunk 2]
```diff
@@ -110,14 +102,17 @@ void shouldThrowExceptionWhenSearchingForNvaPublicationByCristinIdentifierReturn
...
+
+        var actualReport = extractActualReportFromS3Client(eventReference,
+                                                           ParentPublicationException.class.getSimpleName(),
+                                                           childPublication.getIdentifier().toString());
+        assertThat(actualReport.getException(), containsString(INVALID_PARENT_MESSAGE));
     }
 
     @Test
-    void shouldThrowExceptionWhenSearchingForNvaPublicationByCristinIdentifierReturnsNoPublication()
+    void shouldStoreErrorReportWhenSearchingForNvaPublicationByCristinIdentifierReturnsNoPublication()
```
Comment (torarnet) [inline cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinPatchEventConsumerTest.java:115]: NoPublication indicates invalid parent?

File: s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/SqsBatchMessenger.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        this.client = client;
+        this.queue = queue;
+        this.requestIdToMessageBody = new HashMap<>();
+    }
+
+    public PutSqsMessageResult sendMessages(List<EventReference> messageBodies) {
+        var messageRequestEntries = createMessageRequestEntries(messageBodies);
+        var batches = batchEventReferences(messageRequestEntries);
+        var batchRequest = batches.stream().map(this::mapToSendMessageBatchRequestEntry);
+        return batchRequest.map(this::sendToSqs).reduce(PutSqsMessageResult::combine).orElseThrow();
```
Comment (truhacevkir) [inline s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/SqsBatchMessenger.java:41]: Bare forslag:

return batches.stream().map(this::mapToSendMessageBatchRequestEntry)
                               .map(this::sendToSqs)
                               .reduce(PutSqsMessageResult::combine)
                               .orElseThrow();

File: s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/SqsBatchMessenger.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                               .collect(
+                                   Collectors.toList()));
+        return result;
+    }
+
+    private PutSqsMessageResult createPutSqsMessageResult(SendMessageBatchResult response) {
+        var result = new PutSqsMessageResult();
+        var failures =
+            response.getFailed().stream()
+                .map(s -> new PutSqsMessageResultFailureEntry(requestIdToMessageBody.get(s.getId()), s.getMessage()))
```
Comment (truhacevkir) [inline s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/SqsBatchMessenger.java:69]: Ikke så intuitivt hva "s" er, kanskje rename?

File: s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/SqsBatchMessenger.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        sendMessageBatchRequest.withEntries(batchedEvents);
+        sendMessageBatchRequest.withQueueUrl(queue);
+        return sendMessageBatchRequest;
+    }
+
+    private Collection<List<SendMessageBatchRequestEntry>> batchEventReferences(
+        Stream<SendMessageBatchRequestEntry> eventReferenceStream) {
+        var counter = new AtomicInteger();
+        return eventReferenceStream.collect(
+                Collectors.groupingBy(it -> counter.getAndIncrement() / MAX_NUMBER_OF_MESSAGES_PER_BATCH_ALLOWED_BY_AWS))
```
Comment (truhacevkir) [inline s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/SqsBatchMessenger.java:97]: item?

**Review by truhacevkir — APPROVED** [review]

File: template.yaml [hunk 6]
```diff
@@ -383,7 +383,7 @@ Resources:
                 - lambda.amazonaws.com
             Action: sts:AssumeRole
 
-  DefaultLambdaPermissions:
+  LogGroupPermissions:
```
[the remark this answers is not in the document]
Comment (anetteOlli) [reply template.yaml]: Change back. Had a bug where the DefaultLambdaPermissions suddenly only contained s3 permissions. Ketil couldn't figure it out either. So ended up renaming it in order to force it to be recreated.

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java (hunk 1 above)
Comment (anetteOlli) [reply cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumerTest.java]: Thx. Changed to var several places

File: cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinPatchEventConsumerTest.java (hunk 2 above)
Comment (anetteOlli) [reply cristin-import/src/test/java/no/unit/nva/cristin/lambda/CristinPatchEventConsumerTest.java]: renamed to: shouldStoreErrorReportWhenSearchingForParentPublicationByCristinIdentifierReturnsNoPublication

File: s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/FileEntriesEventEmitter.java [hunk 7]
```diff
@@ -53,74 +48,83 @@
...
-        return returnNothingOrThrowExceptionWhenEmissionFailedCompletely(failedEntries);
-    }
-    
-    private boolean thereAreFailures(Try<List<PutEventsResult>> failedEntries) {
-        return completeEmissionFailure(failedEntries) || partialEmissionFailure(failedEntries);
-    }
-    
-    private boolean partialEmissionFailure(Try<List<PutEventsResult>> failedEntries) {
-        return failedEntries.isSuccess() && !failedEntries.orElseThrow().isEmpty();
+        return attemptTPlaceMessagesOnQueue(input);
```
[the remark this answers is not in the document]
Comment (anetteOlli) [reply s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/FileEntriesEventEmitter.java]: fixed

File: s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/SqsBatchMessenger.java (hunk 4 above)
Comment (anetteOlli) [reply s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/SqsBatchMessenger.java]: fixed

File: s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/SqsBatchMessenger.java (hunk 5 above)
Comment (anetteOlli) [reply s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/SqsBatchMessenger.java]: fixed

**Pull request merged**